src/Tribe/Admin/Troubleshooting.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+/**
+ * Admin Troubleshooting for TEC plugins.
+ *
+ * @since TBD
+ *
+ * @package Tribe\Admin
+ */
+
+namespace Tribe\Admin;
+use \Tribe__Settings;
+use \Tribe__Main;
+use \Tribe__Admin__Helpers;
+use \Tribe__Timezones as Timezones;
+
+/**
+ * Class Admin Troubleshooting.
+ *
+ * @since TBD
+ *
+ * @package Tribe\Admin
+ */
+class Troubleshooting
+{
+    /**
+     * Slug of the WP admin menu item
+     *
+     * @since TBD
+     *
+     * @var string
+     */
+    const MENU_SLUG = 'tribe-troubleshooting';
+
+    /**
+     * The slug for the new admin page
+     *
+     * @var string
+     */
+    private $admin_page = null;
+
+    /**
+     * Class constructor
+     */
+    public function hook()
+    {
+        add_action('admin_menu', [ $this, 'add_menu_page' ], 90);
+        add_action('wp_before_admin_bar_render', [ $this, 'add_toolbar_item' ], 20);
+    }
+
+    /**
+     * Adds the page to the admin menu
+     */
+    public function add_menu_page()
+    {
+        if (! Tribe__Settings::instance()->should_setup_pages()) {
+            return;
+        }
+
+        $page_title = esc_html__('Troubleshooting', 'tribe-common');
+        $menu_title = esc_html__('Troubleshooting', 'tribe-common');
+        $capability = apply_filters('tribe_events_troubleshooting_capability', 'install_plugins');
+
+        $where = Tribe__Settings::instance()->get_parent_slug();
+
+        $this->admin_page = add_submenu_page(
+            $where,
+            $page_title,
+            $menu_title,
+            $capability,
+            self::MENU_SLUG,
+            [
+                $this,
+                'do_menu_page',
+            ]
+        );
+    }
+
+    /**
+     * Adds a link to the the WP admin bar
+     */
+    public function add_toolbar_item()
+    {
+        $capability = apply_filters('tribe_events_troubleshooting_capability', 'install_plugins');
+
+        // prevent users who cannot install plugins from seeing addons link
+        if (current_user_can($capability)) {
```

Comment:
spacing standards are like this:
`if ( current_user_can( $capability ) )`

Reply:
Spacing issue is fixed